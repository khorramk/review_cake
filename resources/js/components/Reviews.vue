<template>
        <div class="card-body reviews-card__body">
            <div class="card reviews-card__single-review__wrapper">
                <div class="card-body reviews-card__single-review">
                    {{this.$props.review.reviews}}
                    <router-link class="review-card__single-review_add-comment" :to="`/comments/create/${$props.review.id}`">ðŸ—¨</router-link>
                    <router-link class="review-card__single-review_edit-review" :to="`/reviews/${$props.review.id}/edit`">ðŸ–‰</router-link>
                    <form  class="review-card__single-review_remove-review" v-on:submit.prevent="remove($props.review.id)">
                        <button :disabled="isDisable" type="submit" class="button-remove">â¨¯</button>
                    </form>
                </div>
            </div>
            <CommentsComponent v-for="(comment, i) in $props.comments" :key="i"  :comment="comment" :review-id="$props.review.id" v-model="loading"/>
        </div>        
</template>
<script>
import CommentsComponent from "./CommentsComponent";
export default {
    data() {
        return {
            error: false,
            isDisable: false,
            loading: true,
            
        }
    },
    props: {
        review: {
            default() {
                return {}
            },
            type: Object
        },
        comments: {
            default(){
                return Array(0);
            },
            type: Array
        }
    },
    methods: {
        remove(id){
            this.isDisable = true;
            axios.delete(`/api/reviews/${id}`).then(()=> {
                window.location.href= '/reviews';
            });       
        },
    },
    components: {
        CommentsComponent
    },
    
};
</script>
