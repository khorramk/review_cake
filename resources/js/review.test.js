//test.js

import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import Reviews from './components/Reviews';

const vm = new Vue(Reviews)
test('test comment data after fetching', () => {
    vm.$nextTick().then(()=> {
        expect(vm.comments).not.toHaveLength(0);
    }).catch().finally()
    
});


test('test error after request  ', () => {
    vm.$nextTick().then().catch(()=> {
        expect(vm.error).tobe(true)
    });
    
});

test('disable the isdisable key for button-remove', () => {
    vm.$on('submit', () => {
        expect(vm.isDisable).toBe(true);
    });
});

test('loading property to be true upon fetching ', () => {
    vm.$nextTick().then().catch().finally(()=> {
        expect(vm.loading).toBe(false);
    });
});

it('renders the elements', () => {
    const vm = new Vue(Reviews);
    vm.$nextTick().then(()=> {
 
     expect(vm).toMatchSnapshot();
    }).catch().finally();
  })