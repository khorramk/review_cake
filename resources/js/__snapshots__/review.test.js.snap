// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the elements 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "_propKeys": Array [
      "review",
      "comments",
    ],
    "components": Object {
      "CommentsComponent": Object {
        "data": [Function],
        "methods": Object {
          "submit": [Function],
        },
        "props": Object {
          "comment": Object {
            "type": [Function],
          },
          "reviewId": Object {
            "type": [Function],
          },
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
    },
    "data": [Function],
    "directives": Object {},
    "filters": Object {},
    "methods": Object {
      "remove": [Function],
    },
    "props": Object {
      "comments": Object {
        "default": [Function],
        "type": [Function],
      },
      "review": Object {
        "default": [Function],
        "type": [Function],
      },
    },
    "render": [Function],
    "staticRenderFns": Array [],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {
    "error": false,
    "isDisable": false,
    "loading": true,
  },
  "_directInactive": false,
  "_events": Object {},
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": false,
  "_isVue": true,
  "_props": Object {
    "comments": Array [],
    "review": Object {},
  },
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "_propKeys": Array [
        "review",
        "comments",
      ],
      "components": Object {
        "CommentsComponent": Object {
          "data": [Function],
          "methods": Object {
            "submit": [Function],
          },
          "props": Object {
            "comment": Object {
              "type": [Function],
            },
            "reviewId": Object {
              "type": [Function],
            },
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
      },
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "methods": Object {
        "remove": [Function],
      },
      "props": Object {
        "comments": Object {
          "default": [Function],
          "type": [Function],
        },
        "review": Object {
          "default": [Function],
          "type": [Function],
        },
      },
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {
      "error": false,
      "isDisable": false,
      "loading": true,
    },
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": false,
    "_isVue": true,
    "_props": Object {
      "comments": Array [],
      "review": Object {},
    },
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 1,
    "_vnode": null,
    "_watcher": null,
    "_watchers": Array [],
    "comments": Array [],
    "error": false,
    "isDisable": false,
    "loading": true,
    "remove": [Function],
    "review": Object {},
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 1,
  "_vnode": null,
  "_watcher": null,
  "_watchers": Array [],
  "comments": Array [],
  "error": false,
  "isDisable": false,
  "loading": true,
  "remove": [Function],
  "review": Object {},
}
`;
